src/sagas/wallet.js

```diff
@@ -447,6 +448,43 @@ export function* listenForWalletReady(wallet) {
   }
 }
 
+/**
+ * An abstraction effect to call wallet's checkAddressMine method.
+ * This effect applies a retry strategy and throws a fatal error
+ * after max retry attempts.
+ *
+ * @param {Object} wallet Either fullnode or wallet-service wallet.
+ * @param {string[]} txAddresses A collection of addresses to check.
+ * @returns {{ [address: string]: boolean }} A map of type address:isMine.
+ */
+function* checkAddressMine(wallet, txAddresses) {
+  const maxRetries = 5;
+  const resultHandler = async (cb) => {
+    try {
+      const success = await cb();
+      return { success };
+    } catch (error) {
+      return { error }
+    }
+  };
+
+  let result = null;
+  for (let i = 0; i < maxRetries; i++) {
+    result = yield call(resultHandler, async () => await wallet.checkAddressesMine.bind(wallet)([...txAddresses]));
+    if (result.success) {
+      return result.success;
+    }
+    // attemp 0: 300ms
+    // attemp 1: 330ms
+    // attemp 2: 420ms
+    // attemp 3: 570ms
+    // attemp 4: 780ms
+    yield delay(300 + 30 * (i ** i));
+  }
+
+  yield put(onExceptionCaptured(result.error, true));
+}
+
 export function* handleTx(action) {
```

Comment:
So, from what I understood from the issue, this gets called while the phone is in background so the request fails and throws, is that correct?

Add a comment here explaining this potential issue

Reply:
What happens is simple, the client request the wallet-service and a network issue happen, but the error is not handled. It do not happens all the time a wallet is on background, just sometimes.

Reply:
I see, ok